recipes/antlr4-cppruntime/all/conanfile.py

```diff
@@ -54,28 +63,54 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version],
                   destination=self._source_subfolder, strip_root=True)
 
+    def build_requirements(self):
+        if self.settings.os in ("FreeBSD", "Linux"):
+            self.build_requires("pkgconf/1.7.4")
+
     def requirements(self):
         self.requires("utfcpp/3.2.1")
         if self.settings.os in ("FreeBSD", "Linux"):
             self.requires("libuuid/1.0.3")
 
     def validate(self):
         if str(self.settings.arch).startswith("arm"):
-            raise ConanInvalidConfiguration(f"arm architectures are not supported")
+            raise ConanInvalidConfiguration("arm architectures are not supported")
             # Need to deal with missing libuuid on Arm.
             # So far ANTLR delivers macOS binary package.
-        compiler, version = self.settings.compiler, tools.Version(self.settings.compiler.version)
-        if compiler == "Visual Studio" and version < "16":
-            raise ConanInvalidConfiguration(f"library claims C2668 'Ambiguous call to overloaded function'")
+
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(self.settings.compiler.version)
+        antlr_version = tools.Version(self.version)
+
+        if compiler == "Visual Studio" and compiler_version < "16":
+            raise ConanInvalidConfiguration("library claims C2668 'Ambiguous call to overloaded function'")
             # Compilation of this library on version 15 claims C2668 Error.
             # This could be Bogus error or malformed Antl4 libary.
             # Version 16 compiles this code correctly.
 
+        if antlr_version != "4.9.3":
```

Comment:
```suggestion
        if antlr_version >= "4.10":
```

Probably other `4.9.x` or lower version would work with C++11 as well.

Reply:
Agree - If someone will try adapt previous versions for some reason that should help.